### PR TITLE
fix: crash in ResourceMetrics RPC handling concurrent writers

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
fix: crash in ResourceMetrics RPC handling concurrent writers

## Motivation and Context
Continuation of #19103 that had fixed the crash in 
peer metrics for cluster endpoint.

## How to test this PR?
A crash is observed in a distributed setup when
resource metrics are scraped via the Prometheus endpoint.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression yes since #19004
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by dvaldivia — APPROVED** [review]
Tested. LGTM

File: cmd/peer-rest-client.go [hunk 1]
```diff
@@ -661,6 +641,28 @@ func (client *peerRESTClient) MonitorBandwidth(ctx context.Context, buckets []st
...
+		defer close(ch)
+		for _, m := range resp.Value() {
+			if m == nil {
+				continue
+			}
+			select {
+			case <-ctx.Done():
+				return
+			case ch <- *m:
+			}
```
Comment (klauspost) [inline cmd/peer-rest-client.go:660]: @harshavardhana Please revert this. You *must* consume everything on the response channel.

You are leaking goroutines when canceled.  Why did you change the working implementation?
